src/App.tsx

```diff
@@ -15,48 +15,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SortType = {
+  NONE: 0,
+  ALPHABET: 1,
+  LENGTH: 2,
+};
+
 export const App: React.FC = () => {
+  const [copyArrayList, setCopyArrayList] = useState([...goodsFromServer]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const getReorderedGoods = (param: string) => {
+    if (param === 'alphabetical') {
+      setCopyArrayList([...copyArrayList].sort((a, b) => a.localeCompare(b)));
+      setSortType(SortType.ALPHABET);
+    }
+
+    if (param === 'length') {
+      setCopyArrayList([...copyArrayList].sort((a, b) => a.length - b.length));
+      setSortType(SortType.LENGTH);
+    }
+
+    if (param === 'reverse') {
+      setCopyArrayList([...copyArrayList].reverse());
+      setIsReversed(true);
+    }
+
+    if (param === 'reset') {
+      setCopyArrayList([...goodsFromServer]);
+      setIsReversed(false);
+      setSortType(SortType.NONE);
+    }
+  };
+
+  const AlpabetClassName
+    = (sortType === SortType.ALPHABET && !isReversed)
+    || (sortType === SortType.ALPHABET && isReversed)
+      ? 'button is-info'
+      : 'button is-info is-light';
```

Comment:
You can use `classNames` to set className value, e.g.: `classNames('button', 'is-info', {'is-light': sortType !== SortType.ALPHABET,})` and do it directly inside className={...} of your button instead of creating separate consts